src/lib.rs

```diff
@@ -160,6 +172,10 @@ pub struct Config {
     #[serde(default)]
     /// Configuration for the docker network.
     pub docker_config: DockerConfig,
+    /// FIXME: not sure if this should just be a boolean
+    /// The mxid of an admin user to create, provided as an access token
+    /// to run.
+    admin_user: Option<String>,
```

Comment:
I think that we should have a field `users: Vec<User>`, with

```rust
struct User {
  /// Create user as admin?
  #[serde(Default)]
  admin: bool,

  localname: String,

  // if the password isn't provided, use the localname
  #[serde(Default)]
  password: Option<String>,

  // ... maybe in future versions, a list of rooms to invite/join
}
```

With this, we should be able to avoid storage. We just need to create the users towards the end of `mx-tester up`. If the error states that the username already exists, check that we can login with the password. If we can't or in case of other errors, bailout.

This will probably be more robust than storage.

Reply:
Why do we need the tokens if we have username/password?

Reply:
Yeah I realised pretty quickly you were right about that